be/src/exec/workgroup/scan_task_queue.h

```diff
@@ -67,10 +73,56 @@ class ScanTaskQueue {
     virtual size_t size() const = 0;
     bool empty() const { return size() == 0; }
 
-    virtual void update_statistics(WorkGroup* wg, int64_t runtime_ns) = 0;
+    virtual void update_statistics(ScanTask& task, int64_t runtime_ns) = 0;
     virtual bool should_yield(const WorkGroup* wg, int64_t unaccounted_runtime_ns) const = 0;
 };
 
+class MultiLevelFeedScanTaskQueue final : public ScanTaskQueue {
```

Comment:
the code of mlfq is duplicate, can we extract this code as a utility object for re-use?

Reply:
MLFQ driver queue has some other fields and functions to cancel quickly, such as `std::queue<DriverRawPtr> pending_cancel_queue`.

Reply:
Maybe we could refactor them later.